skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+                 disable_drift_detection=False,
+                 disable_weighted_vote=False,
+                 lambda_value=6,
+                 evaluator_method_metric='acc',
+                 drift_detection_method: BaseDriftDetector=ADWIN(0.001),
+                 warning_detection_method: BaseDriftDetector=ADWIN(0.01),
+                 nominal_attributes=None):
+        """AdaptiveRandomForest class constructor."""
+        super().__init__()          
+        self.nb_ensemble = nb_ensemble        
+        self.feature_mode = feature_mode
+        self.subspace_size = subspace_size
+        self.disable_background_learner = disable_background_learner   
+        self.disable_drift_detection = disable_drift_detection        
+        self.disable_weighted_vote = disable_weighted_vote
+        self.lambda_value = lambda_value
+        self.drift_detection_method = drift_detection_method
+        self.warning_detection_method = warning_detection_method
+        self.instances_seen = 0
+        self._train_weight_seen_by_model = 0.0
+        self.ensemble = None
+        self.nominal_attributes = nominal_attributes
+        self._evaluator_method = ARFBaseClassifierEvaluator  # TODO use skmultiflow evaluator
+
+    def fit(self, X, y, classes=None, weight=None):
+        raise NotImplementedError
+    
+    def partial_fit(self, X, y, classes=None, weight=None):
+        if y is not None:
+            if weight is None:
+                weight = INSTANCE_WEIGHT
+            row_cnt, _ = get_dimensions(X)
+            wrow_cnt, _ = get_dimensions(weight)
+            if row_cnt != wrow_cnt:
+                weight = [weight[0]] * row_cnt                
+            for i in range(row_cnt):
+                if weight[i] != 0.0:
+                    self._train_weight_seen_by_model += weight[i]
+                    self._partial_fit(X[i], y[i], weight[i])
+        
+    def _partial_fit(self, X, y, weight):
+        self.instances_seen += 1
+        
+        if self.ensemble is None:
+            self.init_ensemble(X)
+
+        for i in range(self.nb_ensemble):
+            y_predicted = self.ensemble[i].predict(np.asarray([X]))
+            self.ensemble[i].evaluator.update(y_predicted, np.asarray([y]), weight)
+            k = np.random.poisson(self.lambda_value)
+            if k > 0:
+                self.ensemble[i].partial_fit(np.asarray([X]), np.asarray([y]), np.asarray([k]), self.instances_seen)
+    
+    def predict(self, X):
+        """Predicts the label of the X instance(s)
+        Parameters
+        ----------
+        X: numpy.ndarray of shape (n_samples, n_features)
+            Samples for which we want to predict the labels.
+        Returns
+        -------
+        list
+            Predicted labels for all instances in X.
+        """
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i])
+            if votes == {}:
```

Comment:
Probably this should raise an exception?

Reply:
No, this is to account for prequential evaluation when predictions could be requested before a learner is initialized.